src/pybind/mgr/zabbix/module.py

```diff
@@ -294,6 +326,60 @@ def send(self):
 
         return False
 
+    def discovery(self):
+        osd_map = self.get('osd_map')
+        osd_map_crush = self.get('osd_map_crush')
+
+        # Discovering ceph pools
+        pool_discovery = {
+            pool['pool_name']: step['item_name']
+            for pool in osd_map['pools']
+            for rule in osd_map_crush['rules'] if rule['rule_id'] == pool['crush_rule']
+            for step in rule['steps'] if step['op'] == "take"
+        }
+        pools_discovery_data = {"data": [
+            {
+                "{#POOL}": pool,
+                "{#CRUSH_RULE}": rule
+            }
+            for pool, rule in pool_discovery.items()
+        ]}
+
+        # Discovering OSDs
+        # Getting hosts for found crush rules
+        osd_roots = {
+            step['item_name']: [
+                item['id']
+                for item in root_bucket['items']
+            ]
+            for rule in osd_map_crush['rules']
+            for step in rule['steps'] if step['op'] == "take"
+            for root_bucket in osd_map_crush['buckets']
+            if root_bucket['id'] == step['item']
+        }
+        # Getting osds for hosts with map to crush_rule
+        osd_discovery = {
+            item['id']: crush_rule
+            for crush_rule, roots in osd_roots.items()
+            for root in roots
+            for bucket in osd_map_crush['buckets']
+            if bucket['id'] == root
+            for item in bucket['items']
+        }
+        osd_discovery_data = {"data": [
+            {
+                "{#OSD}": osd,
+                "{#CRUSH_RULE}": rule
+            }
+            for osd, rule in osd_discovery.items()
+        ]}
+        # Preparing recieved data for sending
+        data = {
+            "zabbix.pool.discovery": json.dumps(pools_discovery_data),
```

Comment:
Is there a reason to use `json.dumps`? would it be possible to just return `pools_discovery_data` here?
```suggestion
            "zabbix.pool.discovery": pools_discovery_data,
```

Reply:
Unfortunately zabbix requires for discovery output to be a valid json.